### ISSUE TITLE
No active stream links found

### ISSUE THREAD

**Issue by apexmod2** [Coombaa/TTAutoRecord#46]
Trying everything but I seem to not to be able to run the program correctly. I followed the guide and installed all prerequisites but still get ,"No active stream links found".  I am able to authenticate my account using the cookies but that's as far as it gets. (see screenshot) I have tried Windows 11, Windows 10, and Ubuntu. I have also tried using multiple different account methods such as email/phone login, google login, and twitter/x login. Tried using different versions of the program too. I made sure on all the account I was following accounts that were actively live streaming but that didn't seen to help. What am I doing wrong?
![image](https://github.com/user-attachments/assets/ef3c89b4-faad-40f2-99d6-2eb205658558)

**Comment by nmro97** [comment]
did u change tiktok to english ?

**Comment by nmro97** [comment]
AND as i see ure not using the newest version ure using v4.1.0 -

**Comment by apexmod2** [comment]
Yes language is set to English and I just happened to be using an older version in the screenshot but I have also tried the latest version too.

**Comment by FabLeKebab** [comment]
I think the problem is with Firefox, as an error message appears: 'The page is unavailable. We are sorry! Please try again later.' Then, on TTAutoRecord, it says: 'No active stream links found'

**Comment by nmro97** [comment]
same problem for me  same error in firefox since today

**Comment by FabLeKebab** [comment]
It's happening to me now with Chrome and Firefox, I think they are updating again :s

**Comment by jean75000** [comment]
same for me today :(, I updated Firefox and Chrome, I relaunched and it works :)

quick question, do you know why the project was made with Firefox as the browser and not Chrome?

**Comment by FabLeKebab** [comment]
It's all good for me too, everything restarted without any issues! As for Mozilla, it's probably because Geckodriver is designed to be compatible with that browser (I suppose)

**Comment by jean75000** [comment]
yes but for chrome there is chromedriver and I find that tiktok works much better with chrome (faster to load live) less display bugs

**Comment by jean75000** [comment]
I have this problem 
![image](https://github.com/user-attachments/assets/79a19ec7-f4d6-403b-acea-ed4c210f8f08)

**Comment by Coombaa** [comment]
> same for me today :(, I updated Firefox and Chrome, I relaunched and it works for the moment :)
> 
> quick question, do you know why the project was made with Firefox as the browser and not Chrome?

I cant remember exactly now, I initially did it with chrome and was having issues but it was a while ago now. Feel free to do a PR if you can get it working well with chrome.

Recently they have been killing off sessions often. If you get no active links then you probably got logged out and will need to get fresh cookies.

Ideally it should log back in automatically when this happens with stored credentials, however it doesn't like the automated browser.

**Comment by jean75000** [comment]
Thanks for your answer, I'm not good enough for that :( but I often use Tiktok on firefox and Chrome and I see that the pages and lives load faster with chrome

yes I think so because after several months of testing (the script runs 24/7 on a dedicated server) I noticed that Tiktok randomly disconnects my account (I kept 16 days connected at most)
I have to monitor my server and reconnect regularly

@Coombaa for me the two parts to improve:

- use a stealh system for the Bot so that it is less detectable by tiktok, I know that on the chrome side for example there is selenium-stealth

- monitoring with a telegram bot for example in case we encounter the problem No active stream links found.

**Comment by Coombaa** [comment]
I am working on it now, i have converted it to undetected_chromedriver and also removed the cookie file bs. It now uses your main browser session which should fix OPs issue.

Next i will add a telegram notification, however it will require the user to set up their own bot through botfather

**Comment by jean75000** [comment]
Cool Thanks !!!

**Comment by jean75000** [comment]
> I am working on it now, i have converted it to undetected_chromedriver and also removed the cookie file bs. It now uses your main browser session which should fix OPs issue.
> 
> Next i will add a telegram notification, however it will require the user to set up their own bot through botfather

@Coombaa I saw that there was a version 4.2 of the program, is it usable as is?

**Comment by FabLeKebab** [comment]
> > I am working on it now, i have converted it to undetected_chromedriver and also removed the cookie file bs. It now uses your main browser session which should fix OPs issue.
> > Next i will add a telegram notification, however it will require the user to set up their own bot through botfather
> 
> @Coombaa I saw that there was a version 4.2 of the program, is it usable as is?

![image](https://github.com/user-attachments/assets/e0f037f6-e1c8-498d-901b-05765782b81d)
I’ve just launched version 4.2.0, and so far, everything is running smoothly. You just need to install Chromedriver using the requirements file, then update Chrome to version 130. To do that, click on the three dots in the top-right corner of the browser, go to “Help” at the bottom, and then select “Update"

You can have both versions side by side, meaning you can keep the old version (4.1.2) alongside the new version (4.2.0). Then, it’s necessary to run pip install, which will install Chromedriver for the version necessary

**Comment by jean75000** [comment]
This version now uses Chrome and no longer Firefox, is that it?

**Comment by nmro97** [comment]
> This version now uses Chrome and no longer Firefox, is that it?

the problem was that tiktok logged you out after some time so this one should work better i gues

**Comment by FabLeKebab** [comment]
> Cette version utilise désormais Chrome et non plus Firefox, c'est ça ? Si tel est le cas, GeckoDriver n'est normalement plus nécessaire, il est réservé à Firefox. Sur Chrome, ChromeDriver est celui à utiliser

Yes, it's indeed ChromeDriver. I thought I had made a mistake, so I made a correction, but I ended up making another mistake while correcting, haha. For now, since I sent the message, everything is working very well. And indeed, I no longer need Firefox; it's much better!

**Comment by FabLeKebab** [comment]
> > Cette version utilise désormais Chrome et non plus Firefox, c'est ça ?
> 
> le problème était que tiktok vous a déconnecté après un certain temps, donc celui-ci devrait mieux fonctionner, je suppose

Yes, that's right, TikTok disconnected me three times in one day, but since then, I haven't had any more disconnections. I prefer to start directly from version 4.2.0 and update it gradually, and then configure my bot on Telegram

**Comment by nmro97** [comment]
i have the error could not detect version_main.therefore we are assuming it is chrome 108 or higher    how do i fix this

**Comment by FabLeKebab** [comment]
Honestly, Firefox really got on my nerves because I couldn't do anything anymore. But even on Chrome, there's still an issue: for now, you can't subscribe (and it's been like that for a while). I don't know if it's a general problem or just me, but it's pretty annoying...

**Comment by nmro97** [comment]
i cant use the new version its says no stream link found idk why

**Comment by FabLeKebab** [comment]
> i have the error could not detect version_main.therefore we are assuming it is chrome 108 or higher    how do i fix this 

Even if you let it run, does it keep showing the same error on a loop? Because I get that message too, but it’s more of a warning. It only shows up at the start, and then it works fine afterwards (as you can see in the screenshot I posted above) ?

**Comment by nmro97** [comment]
> > i have the error could not detect version_main.therefore we are assuming it is chrome 108 or higher    how do i fix this
> 
> Even if you let it run, does it keep showing the same error on a loop? Because I get that message too, but it’s more of a warning. It only shows up at the start, and then it works fine afterwards (as you can see in the screenshot I posted above) ?

it says no active stream link found

**Comment by FabLeKebab** [comment]
> i cant use the new version its says no stream link found idk why  

They probably updated TikTok in your region, but not yet on our side. They must be doing it in waves, so it’s possible that we’ll experience these errors soon too...

**Comment by nmro97** [comment]
> > i cant use the new version its says no stream link found idk why
> 
> They probably updated TikTok in your region, but not yet on our side. They must be doing it in waves, so it’s possible that we’ll experience these errors soon too...

i dont think so the main version with firefox works fine for me

**Comment by nmro97** [comment]
so i installed the requirements file add the cookies from tiktok/live , do i need to do anything else ?

**Comment by FabLeKebab** [comment]
> so i installed the requirements file add the cookies from tiktok/live , do i need to do anything else ?

Normally no, except for updating Chrome to version 130 if you haven't done so already, as mentioned above

**Comment by nmro97** [comment]
> > so i installed the requirements file add the cookies from tiktok/live , do i need to do anything else ?
> 
> Normally no, except for updating Chrome to version 130 if you haven't done so already, as mentioned above

ok solved the problem on chrome it wasnt on english :D

**Comment by jean75000** [comment]
@FabLeKebab it's a bug on Tiktok you have to subscribe on the smartphone application and even there sometimes it doesn't work :(

**Comment by jean75000** [comment]
at one point I was blocked, unable to follow anyone on my phone, it unchecked the follow button every time I clicked on it.

I had to create a 2nd account and I did some tests again some time later (10 days) on my 1st account and it worked again ...

**Comment by FabLeKebab** [comment]
That's what I do too; I use my phone to follow people, even though it's rare for me to follow someone. I don't know if it has to do with following a lot of accounts at once, but in my case, I've never had a problem following anyone on my phone (knock on wood)

**Comment by jean75000** [comment]
@FabLeKebab 

I did everything as instructed it doesn't work

![image](https://github.com/user-attachments/assets/8634535a-8a57-41d2-88a0-5c136173c219)

**Comment by jean75000** [comment]
this error :

    i have the error could not detect version_main.therefore we are assuming it is chrome 108 or higher how do i fix this

I analyzed the script it is missing version_main = 130 in the script

![image](https://github.com/user-attachments/assets/4c6337f9-b3e8-4516-8456-0f7f28935e0e)

it must surely be intentional because I think it must be modified manually each time the version is changed (131,132....) so I suppose that is why the parameter is not present I have not tested but I think that if we put options = uc.ChromeOptions(version_main = 130) there should no longer be the message

**Comment by FabLeKebab** [comment]
![image](https://github.com/user-attachments/assets/29e7bf7c-4fab-463b-90d8-a13e9a43dd87)
I have this, it's very strange, yet it works perfectly fine for other people

You're right @jean75000, I just made some changes and I no longer have the error message 👍

**Comment by jean75000** [comment]
@FabLeKebab try to put the user in the json file live_user.json maybe he lives in a non-EU country, I already had the case and as soon as the user was added after a few seconds (no need to stop and restart the script) it worked well

**Comment by FabLeKebab** [comment]
> @FabLeKebab essayez de mettre l'utilisateur dans le fichier json live_user.json peut-être qu'il habite dans un pays hors UE, j'ai déjà eu le cas et dès que l'utilisateur était ajouté au bout de quelques secondes (pas besoin d'arrêter et de redémarrer le script) il a bien fonctionné

I checked that yesterday, it was indeed already in the live_users file but there was no lock_files file. I ended up unsubscribing, it was spamming the console

**Comment by Coombaa** [comment]
> > @FabLeKebab essayez de mettre l'utilisateur dans le fichier json live_user.json peut-être qu'il habite dans un pays hors UE, j'ai déjà eu le cas et dès que l'utilisateur était ajouté au bout de quelques secondes (pas besoin d'arrêter et de redémarrer le script) il a bien fonctionné
> 
> I checked that yesterday, it was indeed already in the live_users file but there was no lock_files file. I ended up unsubscribing, it was spamming the console

There's not really a way around that unless TikTok fix it, The URL is different for each region with a unique hash so it can't just be manually changed. Sometimes the EU link will start working after 5 or 10 minutes.

I tried to tell the support what is happening as it doesnt even work in the web app player but they are brain dead and wont pass it to the developers. If you mainly follow people in another region then it would be best to create an account within that region, then it will have no issues.

TikTok seems to have 3 that i know of, EU, NA and SG.

**Comment by FabLeKebab** [comment]
> > > @FabLeKebab essayez de mettre l'utilisateur dans le fichier json live_user.json peut-être qu'il habite dans un pays hors UE, j'ai déjà eu le cas et dès que l'utilisateur était ajouté au bout de quelques secondes (pas besoin d'arrêter et de redémarrer le script) il a bien fonctionné
> > 
> > 
> > I checked that yesterday, it was indeed already in the live_users file but there was no lock_files file. I ended up unsubscribing, it was spamming the console
> 
> There's not really a way around that unless TikTok fix it, The URL is different for each region with a unique hash so it can't just be manually changed. Sometimes the EU link will start working after 5 or 10 minutes.
> 
> I tried to tell the support what is happening as it doesnt even work in the web app player but they are brain dead and wont pass it to the developers. If you mainly follow people in another region then it would be best to create an account within that region, then it will have no issues.
> 
> TikTok seems to have 3 that i know of, EU, NA and SG.

That's exactly what I noticed; the link was different from this one from a live session of the EU. I don't remember exactly what changed, but it was very subtle. For example, 'tos-useast2a-avt-0068-euttp' only had a few letters that were different (I can't recall exactly which ones), but it's not a big deal. Thanks for your return!

**Comment by FabLeKebab** [comment]
Hey everyone! Could someone please share the 4.2.0 version of the repo with me? I saw there was a recent update, so I tried to install it... and ended up breaking everything. Now I’m struggling to find the original files, so a huge thanks in advance! 🙏

**Comment by GetorixUn** [comment]
sup, how to get 4.2? i see only 4.1.2 in updates...

**Comment by shadowdragon245** [comment]
> sup, how to get 4.2? i see only 4.1.2 in updates...

https://github.com/Coombaa/TTAutoRecord/tree/4.2.0

**Comment by Ftzem** [comment]
4.2 is not working for me. However one question, do you need to follow, or subscribe. I've seen some people using the subscribe and follow. So  I just wanted to make sure

**Comment by shadowdragon245** [comment]
> 4.2 is not working for me. However one question, do you need to follow, or subscribe. I've seen some people using the subscribe and follow. So I just wanted to make sure

just follow, but  you may need to subscribe if you want to see a sub only live

**Comment by GetorixUn** [comment]
> > sup, how to get 4.2? i see only 4.1.2 in updates...
> 
> https://github.com/Coombaa/TTAutoRecord/tree/4.2.0

seems like maybe not to all 4.2 version... i just see only 4.1.2 in branch releases
